jobs/mysql_restore_job.go

```diff
@@ -168,28 +223,34 @@ func restoreMySQLFromS3(
 	}
 
 	if err := xbcloudCmd.Start(); err != nil {
+		cancel()
 		return errors.Wrap(wrapError(err), "xbcloud start failed")
 	}
 	defer func() {
-		err := xbcloudCmd.Wait()
-		if err == nil {
-			return
-		}
-
-		if rerr != nil {
-			rerr = errors.Wrapf(rerr, "xbcloud wait error: %s", err)
-		} else {
-			rerr = errors.Wrap(wrapError(err), "xbcloud wait failed")
+		if err := xbcloudCmd.Wait(); err != nil {
+			cancel()
+			if rerr != nil {
+				rerr = errors.Wrapf(rerr, "xbcloud wait error: %s", err)
+			} else {
+				rerr = errors.Wrap(wrapError(err), "xbcloud wait failed")
+			}
 		}
 	}()
 
 	if err := xbstreamCmd.Start(); err != nil {
+		cancel()
 		return errors.Wrap(wrapError(err), "xbstream start failed")
 	}
-
-	if err := xbstreamCmd.Wait(); err != nil {
-		return errors.Wrap(wrapError(err), "xbstream wait failed")
-	}
+	defer func() {
```

Comment:
The intention was to have the same "blocks" where you have `start` + `defer wait`, so you can move such blocks of code without breaking the pipeline. I.e. you can copy-paste line 225-238, change change command and insert it somewhere in this job and it will work without any problems.